Lib/test/test_imp.py

```diff
@@ -22,6 +22,18 @@ def requires_load_dynamic(meth):
     return unittest.skipIf(not hasattr(imp, 'load_dynamic'),
                            'imp.load_dynamic() required')(meth)
 
+def requires_create_dynamic(meth):
+    """A decorator for CPython-only tests that rely on imp.create_dynamic().
+
+    On some platforms, CPython might not provide imp.create_dynamic().
+    In such cases and in other implementations, this decorator causes
+    decorated tests to be skipped through the normal unittest mechanism.
+    """
+    meth = support.cpython_only(meth)
+    supported = hasattr(imp, 'create_dynamic')
+    deco = unittest.skipIf(not supported, 'imp.create_dynamic() required')
```

Comment:
I would prefer `skipUnless()`.

But why the `requires_create_dynamic` decorator is needed? It is used just once. Why not just use two decorators directly?
```
@unittest.skipUnless(hasattr(imp, 'create_dynamic'), 'imp.create_dynamic() required')
@support.cpython_only
def test_issue31315(self):
    ...
```
Or use special purposed `support.get_attribute()`:
```
@support.cpython_only
def test_issue31315(self):
    create_dynamic = support.get_attribute(imp, 'create_dynamic')
    ...
```